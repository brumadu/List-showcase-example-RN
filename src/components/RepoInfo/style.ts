import styled from "styled-components/native"


export namespace S {
    export const Container = styled.View`
    flex: 1;
    border-bottom-width: 1px;
    justify-content: space-around;
    `;
    export const ContainerRow = styled.View`
    flex-direction: row;
    justify-content: space-between;
    `;
    export const NameText = styled.Text`
    font-size: 16px;
    font-weight: 600;
    `;
    export const Text = styled.Text`
    font-size: 14px;
    font-weight: 400;
    `;
}