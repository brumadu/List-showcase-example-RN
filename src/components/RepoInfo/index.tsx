import React from "react"
import { S } from "./style"

type repoInfo = {
    nameRepo: String;
    stars: String;
    userRepo: String;
}

function RepoInfo(props: repoInfo) {
    return (
        <S.Container>
            <S.ContainerRow>
                <S.NameText>
                    {props.nameRepo}
                    </S.NameText>
                <S.Text>
                    {props.stars}
                    </S.Text>
            </S.ContainerRow>
                <S.Text>{props.userRepo}
                </S.Text>
        </S.Container>
    )
}

export default RepoInfo;