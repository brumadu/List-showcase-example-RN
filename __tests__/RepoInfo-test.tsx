import React from "react";
import renderer from 'react-test-renderer';
import RepoInfo from "../src/components/RepoInfo";

test('renders correctly', () => {
    const tree = renderer.create(<RepoInfo/>).toJSON();
    expect(tree).toMatchSnapshot();
})
