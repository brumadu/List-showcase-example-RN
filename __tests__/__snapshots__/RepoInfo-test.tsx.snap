// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "borderBottomWidth": 1,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      [
        {
          "fontSize": 14,
          "fontWeight": "400",
        },
      ]
    }
  />
</View>
`;
